compiler/src/error.rs

```diff
@@ -41,6 +42,24 @@ pub enum CompileErrorType {
 }
 
 impl CompileError {
+    pub fn is_indentation_error(&self) -> bool {
+        if let CompileErrorType::Parse(parse) = &self.error {
+            if let ParseErrorType::Lexical(lex) = parse {
+                if let LexicalErrorType::IndentationError = lex {
+                    return true;
+                }
+            } else if let ParseErrorType::UnrecognizedToken(token, expected) = parse {
+                if *token == Tok::Indent {
+                    return true;
```

Comment:
Instead of early return, just use `true` here, and return a boolean in each code path.